modules/apps/frontend-theme/frontend-theme-classic/src/WEB-INF/token-definition.json

```diff
@@ -0,0 +1,319 @@
+{
+	"tokenCategories": [
+		{
+			"label": "general",
+			"name": "general"
+		},
+		{
+			"label": "colors",
+			"name": "colors"
+		},
+		{
+			"label": "typography",
+			"name": "typography"
+		},
+		{
+			"label": "spaces",
+			"name": "spaces"
+		}
+	],
+	"tokenSets": [
+		{
+			"label": "general",
+			"name": "general"
+		},
+		{
+			"label": "utility",
+			"name": "utility"
+		},
+		{
+			"label": "layout",
+			"name": "layout"
+		},
+		{
+			"label": "buttons",
+			"name": "buttons"
+		},
+		{
+			"label": "alerts",
+			"name": "alerts"
+		},
+		{
+			"label": "cards",
+			"name": "cards"
+		},
+		{
+			"label": "forms",
+			"name": "forms"
+		},
+		{
+			"label": "nav",
+			"name": "nav"
+		},
+		{
+			"label": "portlet",
+			"name": "portlet"
+		},
+		{
+			"label": "tables",
+			"name": "tables"
+		},
+		{
+			"label": "lists",
+			"name": "lists"
+		}
+	],
+	"tokens": [
+		{
+            "ccp": "--white",
```

Comment:
So, this is the only thing... @izaera, @ruben-pulido, @marcoscv-work, @wincent... thoughts on how to better model it now that we can see it?

I still think something like:

```
{
    "editorType": "ColorPicker",
    "label": "gray-100",
     "mappings": {
        "ccp": "gray-100"
        "figma": {
            "id": 200,
            "value": "gray-100",
        }
    },
    "name": "gray100Color",
    "tokenCategoryName": "colors",
    "tokenSetName": "general",
    "type": "String"
}
```

We could make the CSS CustomProperties mapping as simple as a string to allow for more complex ones in the future.

We would need to decide if we want `mappings` to be an indexed object as in the example or an array with `mappingType` and `mapping` properties. The second one is more verbose but also more readable...

Thoughts?

Reply:
I prefer indexed objects because:

1. They enforce keys to be unique
2. They let you index by key in O(1)

Said that, it is obvious that we can do whatever we want in the JSON and then transform it internally to suit our programming needs. 

However, I don't know how easy/possible it is to enforce unique keys inside an array in JSON schema. That could be key...

Reply:
I was wondering why we include custom mappings in this file:

1. Because it's their final place and we want them to be here
2. Because we don't know yet what to do with them and we are just proofing

@ruben-pulido seems to be aligned with me (as per private conversation) and thinks the reason is 2

Reply:
> However, I don't know how easy/possible it is to enforce unique keys inside an array in JSON schema. That could be key...

The current specification of JSON schema does not allow it AFAIK. 
There is an open issue asking for [this](https://github.com/json-schema-org/json-schema-vocabularies/issues/22).

Reply:
In other places in the portal where we want to guarantee uniqueness we are adding code to perform that [check](https://github.com/liferay/liferay-portal/blob/49fa14f313f77f5664c0bb7a36017cb06b77ca6d/modules/apps/fragment/fragment-service/src/main/java/com/liferay/fragment/internal/validator/FragmentEntryValidatorImpl.java#L95) outside of the JSON schema validation

Reply:
That's OK, but if we can get it for free... ;-)

Reply:
Ok, so... should we simply do 

```javascript
{
    "editorType": "ColorPicker",
    "label": "gray-100",
     "ccp": "gray-100",
    "name": "gray100Color",
    "tokenCategoryName": "colors",
    "tokenSetName": "general",
    "type": "String"
}
```

And push the ball forward? Figure out based on this if we want to push this to a separate mappings file or a mappings property or what?

Off the top of my head, we shouldn't separate mappings to a different file, because then it will be way too easy for tokens to fall out of sync and have no mappings (adding a token, forgetting the mapping... removing a token, keeping the mapping...). But we can maybe derive this from experience? 🤔

Reply:
Maybe something more descriptive than `ccp` like `cssCustomProperty`? or `cssVar`?

Reply:
> We would need to decide if we want mappings to be an indexed object as in the example or an array with mappingType and mapping properties. The second one is more verbose but also more readable...
I favor the more readable variant with an array with mappingType and mapping properties.

Reply:
> it will be way too easy for tokens to fall out of sync and have no mappings

That's a point too :thinking:

My concern was that the person writing the tokens may not know anything about the mappings. It could even be that the mappings are not known at the stage where the theme is developed. But it is true that that problem may be fixed with a proper development workflow...

Maybe we can settle on adding mappings in the file and then let it be extended (after deployment of the theme) if needed.

In that case, we just need to decide if we want an array or an object...

Reply:
Well... seeing as we have a schema and indexed objects are not spec'd... looks like we'll have an array?

```javascript
{
    "editorType": "ColorPicker",
    "label": "gray-100",
     "mappings": [
          {
               "mappingType": "cssVariable",
               "value": "gray-100"   
        }
    ],
    "name": "gray100Color",
    "tokenCategoryName": "colors",
    "tokenSetName": "general",
    "type": "String"
}
```

This, @Ruben-Pulido, @izaera, @eduardoallegrini? 👆

Reply:
That looks good to me. I'll update the token definition JSON schema in https://github.com/liferay-frontend/liferay-portal/pull/99 accordingly

Reply:
>    "mappings": [
          {
               "mappingType": "cssVariable",
               "value": "gray-100"   
        }
    ],

I'll use `type` instead of `mappingType` as the name of the field. Prefixing `type` but not prefixing `value` seems otherwise inconsistent

Reply:
The example mentioned above will then be:

```
{
    "editorType": "ColorPicker",
    "label": "gray-100",
     "mappings": [
          {
               "type": "cssVariable",
               "value": "gray-100"   
        }
    ],
    "name": "gray100Color",
    "tokenCategoryName": "colors",
    "tokenSetName": "general",
    "type": "String"
}
```
 
@eduardoallegrini Can you modify `token-definition.json` to include the change?